src/core/src/package_managers/PackageManager.py

```diff
@@ -440,15 +440,23 @@ def add_arch_dependencies(self, package_manager, package, packages, package_vers
         """
         Add the packages with same name as that of input parameter package but with different architectures from packages list to the list package_and_dependencies.
         Only required for yum. No-op for apt and zypper.
-        
+
         Parameters:
         package_manager (PackageManager): Package manager used.
         package (string): Input package for which same package name but different architecture need to be added in the list package_and_dependencies.
         packages (List of strings): List of all packages selected by user to install.
         package_versions (List of strings): Versions of packages in packages list.
-        package_and_dependencies (List of strings): List of packages along with dependencies. This function adds packages with same name as input parameter package 
+        package_and_dependencies (List of strings): List of packages along with dependencies. This function adds packages with same name as input parameter package
                                                     but different architecture in this list.
         package_and_dependency_versions (List of strings): Versions of packages in package_and_dependencies.
         """
         pass
 
+    @abstractmethod
+    def set_security_esm_package_status(self, operation):
```

Comment:
This implementation is not ideal, as it's now leaking Ubuntu's implementation requirements into all package manager implementations. Not going to consider this blocking for this PR as it can be refactored later. Some of it may be mitigated in just function naming but TBD.

Reply:
Yes, I felt the same way. Hence was bringing this up in our previous discussions. This is similar to how we have implemented add_arch_dependencies already. 
If this is not the case, we should be adding this to common code under a "APT" check if-block.

Reply:
Marking to resolve later.